src/octoprint/plugins/pluginmanager/static/js/pluginmanager.js

```diff
@@ -2054,9 +2253,210 @@ $(function () {
 
                 self._processPluginManagementResult(data, action, name);
                 self.requestPluginData();
+            } else if (messageType === "queued_installs") {
```

Comment:
I have admittedly not fully walked through all of this, but is there maybe any chance to make some bits here reusable so it looks less duplicated?

Reply:
I'm not sure there is a good way to refactor this. It's all based on different stages of the process and although it may look like a lot of duplication it actually isn't I don't think. For example, on print cancel it shows a different button then on print completion, and timer function, etc. If you see things differently please let me know with a possible hint on direction.